### PR TITLE
Use Jupytext to create index for mybinder deployment

### PR DESCRIPTION
**Pull request by cgranade**
This PR adds a new Markdown file, `binder-index.md`, that can be used as a landing page for mybinder.org deployment. By adding `jupytext` as a Dockerfile dependency, `binder-index.md` appears to Jupyter Notebook as a Q#-language notebook, enabling some really neat experiences. For example, if you follow the "CHSH Game" link below, that takes you to a Jupyter Notebook rendering of that sample's README.md:

image

That README.md is then itself an executable PowerShell-language notebook, such that instructions in the ordinary README can be executed in-place:

image

So far, this PR has been tested in VS Code devcontainers, but once merged in we can test with binder before redirecting the `★binder` branch to point to the new samples landing experience.

### CODE REVIEW

**Comment by guenp** [comment]
@cgranade This is indeed really neat! would it be an idea to add a note on how to use said features to `CONTRIBUTING.md`?

**Comment by cgranade** [comment]
> @cgranade This is indeed really neat! would it be an idea to add a note on how to use said features to `CONTRIBUTING.md`?

That's a good idea, @guenp, totally agreed. I think that should be deferred until after we test with mybinder deployment, however, so that we can make sure any advice that we add there is as accurate as possible.

File: samples/algorithms/database-search/README.md [hunk 1]
```diff
@@ -1,19 +1,61 @@
...
+```
+
+### Running the Sample in Jupyter Notebook
 
 This sample can also be viewed using Jupyter Notebook.
 To do so, ensure that you have the IQ# kernel installed using the instructions in the [getting started guide](https://docs.microsoft.com/quantum/install-guide/jupyter).
 Then, start a new Jupyter Notebook session from this directory:
 
-```Command Line
+```
```
Comment (msoeken) [inline samples/algorithms/database-search/README.md:58]:
```suggestion
```powershell
```

**Review by msoeken — APPROVED** [review]
This is a great overview page. Thanks!

**Pull request merged**